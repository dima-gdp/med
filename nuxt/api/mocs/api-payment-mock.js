export const mockPaymentCreatedDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'created',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}

export const mockCreatePaymentFixture = {
  data: {
    id: 9,
    orderId: 1,
    userId: 32,
    invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
    description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
    paymentMethod: 'bank_card',
    status: 'created',
    order: {
      itemTitle: 'Test',
      itemUrl: '/',
    },
    paidSum: '3000',
    paidCurrency: 'RUB',
    paidAt: null,
    confirmationUrl:
      'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
    createdAt: 1629195418,
    updatedAt: 1629195418,
  },
}

export const mockPaymentFailedDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'failed',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}

export const mockPaymentPaidDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'paid',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}
