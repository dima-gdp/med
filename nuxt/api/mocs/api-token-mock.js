export const mockTokenIdSuccessFixture = {
  id: '237457',
  userId: null,
  type: 'password_reset',
  sendMethod: 'call',
  sendTo: '79115746172',
  isUsed: false,
  usedAt: null,
  expireAt: 1626345855,
  createdAt: 1626259455,
  updatedAt: 1626259455,
}

export const mockTokenIdFailFixture = {
  errors: [
    {
      source: {
        pointer: '/data/attributes/sendTo',
      },
      detail: 'Введите номер в формате 79990001111',
      status: '422',
    },
  ],
}

export const mockValidateSuccessFixture = {
  id: '237457',
  validate: true,
}

export const mockValidateFailFixture = {
  id: '237457',
  validate: false,
}

export const mockPasswordResetSuccessFixture = {
  id: '49',
  userId: 49,
}

export const mockPasswordResetFailFixture = {
  errors: [
    {
      code: '0',
      status: '500',
      title: 'Internal Server Error',
      detail: 'Возникла внутренняя ошибка сервера.',
    },
  ],
}
