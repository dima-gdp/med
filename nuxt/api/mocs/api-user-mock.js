/**
 * @module api-mocs
 */

import { Deserializer } from 'jsonapi-serializer'
const jsonApiOpts = { keyForAttribute: 'camelCase', dataLinks: {} }
const deserializer = new Deserializer(jsonApiOpts)

export const mockUserDataFixture = {
  id: 9,
  email: 'alexandr.anciferov@gmail.com',
  emailValidate: true,
  phoneValidate: true,
  external: false,
  profile: {
    userId: 9,
    positionId: 1,
    companyId: 213,
    firstName: 'Александр',
    middleName: 'Николаевич',
    lastName: 'Анциферов',
    avatar: null,
    id: '9',
  },
}

export const mockUserMenuFixture = [
  {
    type: 'panel',
    visible: true,
  },
  {
    type: 'profile',
    visible: true,
  },
  {
    type: 'subscribe',
    visible: true,
  },
  {
    type: 'certificate',
    visible: false,
  },
  {
    type: 'curator',
    visible: false,
  },
  {
    type: 'course',
    visible: false,
  },
  {
    type: 'orders',
    visible: true,
  },
  {
    type: 'logout',
    visible: true,
  },
]

export const mockPossibleMenuItemsFixture = [
  {
    type: 'panel',
    url: '/admin',
    title: 'Панель управления',
    sort: 1,
  },
  {
    type: 'profile',
    url: '/user',
    title: 'Профиль',
    sort: 2,
  },
  {
    type: 'subscribe',
    url: '/mysubscribe',
    title: 'Подписки',
    sort: 3,
  },
  {
    type: 'orders',
    url: '/orders/list',
    title: 'Мои заказы',
    sort: 7,
  },
  {
    type: 'logout',
    url: '/user/sign-in/logout',
    title: 'Выйти',
    sort: 8,
  },
]

export const mockUserAuthFixture = {
  identity: 'admin@s256.dev',
  token: 'eVLcXHtojhi3AB5tHC0wQSdAzl979qystHGTBysb',
  userId: 1,
}

export async function GET_STATS() {
  const responseData = {
    data: {
      id: 'user',
      type: 'stats',
      attributes: {
        'internal-count': 150940,
      },
    },
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}

export async function GET_USER_AND_PROFILE(userId) {
  const responseData = {
    data: {
      id: '' + userId,
      type: 'user',
      attributes: {
        id: userId,
        email: 'alexandr.anciferov@gmail.com',
        'email-validate': true,
        'phone-validate': true,
        external: false,
      },
      relationships: {
        profile: {
          data: {
            id: '' + userId,
            type: 'profile',
          },
        },
      },
    },
    included: [
      {
        id: '' + userId,
        type: 'profile',
        attributes: {
          'user-id': userId,
          'position-id': 1,
          company_id: 213,
          firstName: 'Александр',
          middleName: 'Николаевич',
          lastName: 'Анциферов',
          avatar: null,
        },
      },
    ],
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}

export async function GET_USER_MENU(userId) {
  const responseData = {
    data: {
      id: '' + userId,
      type: 'menu',
      attributes: {
        id: '' + userId,
        items: [
          {
            type: 'panel',
            visible: true,
          },
          {
            type: 'profile',
            visible: true,
          },
          {
            type: 'subscribe',
            visible: true,
          },
          {
            type: 'certificate',
            visible: true,
          },
          {
            type: 'curator',
            visible: false,
          },
          {
            type: 'course',
            visible: true,
          },
          {
            type: 'logout',
            visible: true,
          },
        ],
      },
    },
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}
