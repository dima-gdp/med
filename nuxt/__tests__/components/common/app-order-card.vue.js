import { mount } from '@vue/test-utils'
import AppOrderCard from '~/components/common/app-order-card'
import { mockOrdersListFixture } from '~/api/mocs/api-order-mock'

describe('~/components/common/app-order-card', () => {
  let wrapper

  const createComponent = (propsData) => {
    wrapper = mount(AppOrderCard, {
      propsData,
    })
  }

  afterEach(() => {
    wrapper.destroy()
  })

  it('Если тип заказа модуль', () => {
    createComponent({
      order: mockOrdersListFixture.initialFormatted.ordersList[0].list[0],
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Если тип заказа итерация курса', () => {
    createComponent({
      order: mockOrdersListFixture.initialFormatted.ordersList[0].list[9],
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
