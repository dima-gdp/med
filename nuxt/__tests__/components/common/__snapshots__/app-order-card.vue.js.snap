// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`~/components/common/app-order-card Если тип заказа итерация курса 1`] = `
<div class="order-card">
  <div class="order-card__image-block"><img src="~/assets/svg/logo.svg" alt="icon"></div>
  <div class="order-card__content-block">
    <div class="order-card__info-block">
      <div class="order-card__type h-6 color--king-blue"><i class="eva order-card__icon-type eva-globe-2"></i>
        Онлайн-курс
      </div>
      <div class="order-card__status card-timer-compress order-card__status--canceled">
        отменён
      </div>
      <div class="order-card__number color--cool-grey card-data-compress">
        Заказ №8
      </div>
    </div>
    <h3 class="order-card__title h-4">Недоношенные дети. Итерация #1</h3>
    <div class="order-card__footer">
      <div class="order-card__price order-card__price--failed h-3">4500 ₽</div>
      <div class="order-card__btn-block">
        <!----> <a href="http://med.s256.xyz/course/nedonosennye-deti" target="_blank" class="t-b-l s-btn s-btn--c-green s-btn--t-ghost s-btn--compressed"><span class="s-btn__content d-f f-c">Подробнее</span></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`~/components/common/app-order-card Если тип заказа модуль 1`] = `
<div class="order-card">
  <div class="order-card__image-block"><img src="~/assets/svg/logo.svg" alt="icon"></div>
  <div class="order-card__content-block">
    <div class="order-card__info-block">
      <div class="order-card__type h-6 color--king-blue"><i class="eva order-card__icon-type eva-monitor-outline"></i>
        Вебинар
      </div>
      <div class="order-card__status card-timer-compress order-card__status--created">
        не оплачен
      </div>
      <div class="order-card__number color--cool-grey card-data-compress">
        Заказ №17
      </div>
    </div>
    <h3 class="order-card__title h-4">Микронутриенты для поддержки нормальной беременности ИЛИ физиологическая беременность - нужна ли нутрициальная поддержка? Зоны риска</h3>
    <div class="order-card__footer">
      <div class="order-card__price order-card__price--failed h-3">12 ₽</div>
      <div class="order-card__btn-block"><button class="t-b-l s-btn s-btn--c-yellow s-btn--t-solid s-btn--compressed"><span class="s-btn__content d-f f-c"><client-only><i class="eva eva-credit-card-outline s-btn__icon s-btn__icon--left"></i></client-only>Оплатить</span></button> <a href="http://med.s256.xyz/module/mikronutrienty-dla-podderzki-normalnoj-beremennosti" target="_blank" class="t-b-l s-btn s-btn--c-green s-btn--t-ghost s-btn--compressed"><span class="s-btn__content d-f f-c">Подробнее</span></a></div>
    </div>
  </div>
</div>
`;
